src/libtrx/include/libtrx/game/savegame/enum.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+typedef enum {
+    SAVEGAME_STAGE_BEFORE_LOAD,
+    SAVEGAME_STAGE_AFTER_LOAD,
+    SAVEGAME_STAGE_BEFORE_SAVE,
+} SAVEGAME_STAGE;
+
+typedef enum {
+    SAVEGAME_FORMAT_INVALID = 0,
+    SAVEGAME_FORMAT_LEGACY = 1,
+    SAVEGAME_FORMAT_BSON = 2,
+} SAVEGAME_FORMAT;
+
+typedef enum {
+    VERSION_LEGACY = -1,
+#if TR_VERSION == 1
+    VERSION_0 = 0,
+    VERSION_1 = 1,
+    VERSION_2 = 2,
+    VERSION_3 = 3,
+    VERSION_4 = 4,
+    VERSION_5 = 5,
+    VERSION_6 = 6,
+
+    // Added extra footer after the compressed BSON structure for quicker
+    // access to essential data, such as the level counter and the level title,
+    // without the need to parse the entire BSON document.
+    // Added TR2+ stats ammo hits/used, health packs used, distance travelled.
+    VERSION_7 = 7,
+
+    // Added the current TR1X version string.
+    VERSION_8 = 8,
+#elif TR_VERSION == 2
+    VERSION_0 = 0,
+#endif
```

Comment:
TBH I'd use linear versioning across both games, considering we're aiming for a common TRX format. WDYT?

Reply:
That makes sense, done 😌